src/pybind/mgr/dashboard/tox.ini

```diff
@@ -176,6 +181,21 @@ commands =
      python3 -m dashboard.controllers.docs {env:OPENAPI_FILE_TMP:{env:OPENAPI_FILE}}
      check: diff {env:OPENAPI_FILE} {env:OPENAPI_FILE_TMP}
 
+[testenv:grafonnet-{check,fix}]
+basepython = python3
+whitelist_externals =
+    jsonnet
+    bash
+description =
+    check: Ensure that auto-generated grafana dashboard files matches the current version
+    fix: generate dashboard json files from jsonnet file with latest changes
+deps =
+    {[grafonnet]deps}
+passenv = GRAFONNET_PATH
+commands =
+     check: bash {env:CEPH_ROOT:../../../..}/monitoring/grafana/dashboards/test-jsonnet.sh
+     fix: jsonnet -m . {env:CEPH_ROOT:../../../..}/monitoring/grafana/dashboards/jsonnet/grafana_dashboards.jsonnet
```

Comment:
this tox should be located under `monitoring/grafana/dashboards`.

Reply:
Yup, why not splitting this and moving it to the monitoring/... dir?

Reply:
@tchaikov . I created a new tox file in the grafana/dashboards directory and moved this tox there. But I couldn't see it running in the make check. Do I need to configure it somewhere else as well?

Reply:
@aaSharma14 if you check the top dir's CMakeLists.txt, you'll see that the `monitoring...` dir is conditionally added if `WITH_GRAFANA` option is enabled (and by default is not):

https://github.com/ceph/ceph/blob/a62770a75292267f9ba33430a64dcc831db1a336/CMakeLists.txt#L685-L688

This is because that was for delivering and installing Grafana Dashboards along with the packages, before cephadm and delivering a Grafana containers with the Dashboard already pushed in.

I'd remove from this CMakeLists.txt the previous code that installed the JSON Dashboards, and simply enable the WITH_GRAFANA option in our run-make-check.sh script.